docs: document UDF configs (#7540)
confluentinc/ksql: docs/reference/server-configuration.md

alongside the ksqlDB server. Don't set this property if you're using
an external `ksql.connect.url`.

## `ksql.extension.dir`

> The directory in which ksqlDB will look for UDFs. The default value

Reviewer: ```suggestion
The directory in which ksqlDB looks for UDFs. The default value
```

> is the `ext` directory relative to the ksqlDB's current working directory.

Reviewer: ```suggestion
is the `ext` directory relative to ksqlDB's current working directory.
```

Reviewer: Is the working directory the same as `$CONFLUENT_HOME/bin`?

Author: I'm actually not sure. By default, it would actually be the directory that you execute the command to start ksqldb. But maybe our startup script change the CWD before launching.

ksql.fail.on.production.error=false
```

## `ksql.functions.<UDF Name>.<UDF Config>`

Makes custom configuration values available to the UDF specified by name.

> For example, if a UDF is named "formula", then you can pass a config

Reviewer: ```suggestion
For example, if a UDF is named "formula", you can pass a config
```

> to that UDF by specifying the property `ksql.functions.formula.base.value`.

Reviewer: ```suggestion
to that UDF by specifying the `ksql.functions.formula.base.value` property.
```

> Within the UDF's `configure` method, it will have access to the propery

Reviewer: ```suggestion
Access the property in the UDF's `configure` method
```

> by the full name, `ksql.funcitons.formula.base.value`. This example

Reviewer: ```suggestion
by using its full name, `ksql.functions.formula.base.value`. This example
```
